test/models/userRefs.ts

```diff
@@ -0,0 +1,18 @@
+import { arrayProp, prop, Ref, Typegoose } from '../../src/typegoose';
+
+export class UserRef extends Typegoose {
+    @prop({ ref: UserRef, default: null })
+    public master: Ref<UserRef>;
+
+    @arrayProp({ itemsRef: UserRef, default: [] })
+    public subAccounts: Ref<UserRef>[];
+
+    @prop({ required: true })
+    public name: string;
+}
+
+export const UserRefModel = new UserRef().getModelForClass(UserRef, {
+    schemaOptions: {
+        collection: 'UserRefs'
```

Comment:
why use a specific collection?

Reply:
because at first the tests had problems with another model instance, but with with the class name changed , now it's fixed, i will clean this